### ISSUE TITLE
learning_rate_control_error_measure should match exactly

### ISSUE THREAD

**Issue by albertz** [rwth-i6/returnn#1146]
Currently when `learning_rate_control_error_measure` (the error key) does not match exactly, it will use some heuristics. See `LearningRateControl.get_error_key`.

This is also because the error key can change depending on whether there is a single loss or multiple, see `LearningRateControl.set_epoch_error`.

This behavior can potentially hide user mistakes like providing an invalid error key.

This was already mentioned in https://github.com/rwth-i6/returnn_common/issues/211 where an option like `learning_rate_control_error_measure_must_match` was suggested to enforce an exact match. However, after some discussion with @mmz33, we think that having a new behavior version might be a another, maybe a better option instead.

Such new behavior (either via new behavior version, or via such option) would need to change both `get_error_key` and `set_epoch_error`.

**Comment by albertz** [comment]
No feedback here?

We can also first introduce such an option and later change the default of this option via a new behavior version. Or maybe also right away, and then people can either disable it to get the old behavior, or adapt it.

**Comment by albertz** [comment]
Despite this, there is also the inconsistency of a single vs multiple losses. In the single case, it would just store `dev_score` etc, and only with multiple losses it appends the loss name. I think this option should also enable to always store the full name.

**Comment by patrick-wilken** [comment]
I see that storing the full name in all cases would require a new option / behaviour version. But with the current loss names, can't we already add a check that if `learning_rate_control_error_measure` is set it should match one of the losses? (Maybe optionally including an "_output" postfix, that's currently in the code.) That is a bug fix rather than a new behaviour, right? Nobody sets this option and expects it to be ignored.
